src/ui/StarRating/StarRating.ts

```diff
@@ -0,0 +1,126 @@
+import 'styling/_StarRating';
+import { $$ } from '../../utils/Dom';
+import { Utils } from '../../utils/Utils';
+import { l } from '../../strings/Strings';
+import { Component } from '../Base/Component';
+import { SVGIcons } from '../../utils/SVGIcons';
+import { exportGlobally } from '../../GlobalExports';
+import { IQueryResult } from '../../rest/QueryResult';
+import { IComponentBindings } from '../Base/ComponentBindings';
+import { ComponentOptions, IFieldOption } from '../Base/ComponentOptions';
+import { Initialization } from '../Base/Initialization';
+
+const DEFAULT_SCALE = 5;
+
+export interface IStarRatingOptions {
+  rating?: IFieldOption;
+  numberOfRatings?: IFieldOption;
+  ratingScale?: number;
+}
+/**
+ * The `StarRating` component renders a 5-star rating widget.
+ *
+ * This component is a result template component (see [Result Templates](https://developers.coveo.com/x/aIGfAQ)).
+ */
+export class StarRating extends Component {
+  static ID = 'StarRating';
+  private rating: number;
+  private numberOfRatings: number;
+  private ratingScale: number;
+
+  static doExport = () => {
+    exportGlobally({
+      StarRating: StarRating
+    });
+  };
+
+  static options: IStarRatingOptions = {
+    /**
+     * Specifies the rating to be displayed as stars.
+     *
+     * Specifying a value for this parameter is required in order for the StarRating component to work.
+     */
+    rating: ComponentOptions.buildFieldOption({ defaultValue: '@rating', required: true }),
+
+    /**
+     * Specifies the number to be displayed in the label indicating the number of ratings.
+     *
+     * If unspecified, no ratings number label is shown. If a value of `0` is provided, a `No Ratings` label is displayed
+     */
+    numberOfRatings: ComponentOptions.buildFieldOption({ required: false }),
```

Comment:
At first, I was adding a comment that it should be a `numberOptions`. 

But then I understood that it displays a value from a field returned by the index. 
So the description could get some love.

Also, I can immediately envision people asking for this to be pulled from a group by result derived from the `rating` field, probably through a facet range request ... 

Because as a user of the library, that's how I would actually expect to retrieve it. 🤔 Maybe I'm wrong.

If we don't wanna add too much complexity here, then it's fine to leave it like that, but unfortunately this seems extremely limited in the current implementation, and I'm not 100% sure how useful it's going to be.

Reply:
All good points. I was very aware while writing it that the wording was bad. :P If documentation has an easy alternative I'm all ears.

This component was built in conjunction with our default b2bcommerce result template to display the rating information, so the current spec is quite limited for now. But I agree and can definitely see it being a facet and filter value like we see on many commerce sites.
If it's possible to have this initial implementation for now and build on it in the near future, that would be good.